source/runtime_gui.cpp

```diff
@@ -1283,7 +1283,19 @@ void reshade::runtime::draw_gui()
 			const std::time_t t = std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
 			struct tm tm; localtime_s(&tm, &t);
 
-			const int temp_size = ImFormatString(temp, sizeof(temp), _clock_format != 0 ? "%02u:%02u:%02u" : "%02u:%02u", tm.tm_hour, tm.tm_min, tm.tm_sec);
+			char temp_clock[32];
+			switch(_clock_format){
+				case 0:
+					sprintf_s(temp_clock, "%02u:%02u", tm.tm_hour, tm.tm_min)
+					break;
+				case 1:
+					sprintf_s(temp_clock, "%02u:%02u:%02u", tm.tm_hour, tm.tm_min, tm.tm_sec)
+					break;
+				case 2:
+					sprintf_s(temp_clock, "%.4d-%.2d-%.2d %02u:%02u:%02u", tm.tm_year + 1900, tm.tm_mon + 1, tm.tm_mday, tm.tm_hour, tm.tm_min, tm.tm_sec);
+					break;
+			}
+			const int temp_size = ImFormatString(temp, sizeof(temp), "%s", temp_clock);
```

Comment:
This can cause a buffer overun if the clock format is not 0, 1 or 2 (which is possible, as this can be overriden in the config file). It's also not necessary to do two string formats here.
Since this is a pretty simple change, I'll just push a commit with this fixed directly.

Reply:
Thanks!